tsl/src/compression/compression.c

```diff
@@ -298,14 +320,48 @@ compress_chunk(Oid in_table, Oid out_table, const ColumnCompressionInfo **column
 
 	TupleDesc in_desc = RelationGetDescr(in_rel);
 	TupleDesc out_desc = RelationGetDescr(out_rel);
+	in_rel_index_iods = RelationGetIndexList(in_rel);
+	if (n_keys == 1)
```

Comment:
Let's make it work for arbitrary number of columns. We have to just loop over them and if the next column doesn't match or the direction is different, don't use the index.

Reply:
The initial PR needs to at least have support for n_keys <= 2 as that will cover most configurations. n_keys = 1 would only apply for compressed hypertables with no segmentby columns.

Reply:
Implemented for multi-column support